sktime/base/_base.py

```diff
@@ -565,46 +565,88 @@ def get_fitted_params(self, deep=True):
                 "fitted yet, please call fit on data before get_fitted_params"
             )
 
-        # collect non-nested fitted params of self
-        fitted_params = self._get_fitted_params()
-
-        # the rest is only for nested parameters
-        # so, if deep=False, we simply return here
         if not deep:
-            return fitted_params
+            return self._gfp_non_nested_plugin()
+
+        fitted_params = {}
+        get_fitted_params_from_plugins = [
+            self._gfp_non_nested_plugin,
+            self._gfp_nested_skbase_plugin,
+            self._gfp_nested_sklearn_plugin,
+            self._gfp_sklearn_pipeline_plugin,
+        ]
+
+        for plugin in get_fitted_params_from_plugins:
```

Comment:
I think this might be the reason for current failures.

The way I would do it is not apply plugins to self, but to children.

That is, you first get the components, via current logic, or possibly via `_components` (perhaps some composites need overrides).

o each component, apply the plugins, and pool fitted params.

Why: the problematic components are those that are not `BaseObject`-s. With the current logic, you always have to deal with components. Whereas, if you apply directly to children, you do not need to deal with the parents.

Reply:
I think that's what we're already doing! (please correct me if you think I'm wrong) The difference is that rather than applying each plugin on the children, we apply it to `self`, but each plugin operates on the inner components directly (as you mentioned via the use of `_components` in case of `skbase` objects and `named_steps` in case of sklearn pipelines).

Reply:
But your suggestion makes complete sense. However, all of the plugins you see here (except for the sklearn pipeline one) are essentially refactors of what we initially had. I've ensured no logic has been changed there. Your suggestion would require reworking the implemented logic and we might have to tackle two tasks (reworking logic to retrieve on components rather than self and enabling visibility of sklearn pipeline params). I suggest we should take on the former point in a separate PR and let this PR be exclusively dedicated to the incorporation of the latter.